client/reader/discover/index.node.js

```diff
@@ -24,5 +41,33 @@ const discoverSsr = ( context, next ) => {
 export default function ( router ) {
 	const anyLangParam = getAnyLanguageRouteParam();
 
-	router( [ '/discover', `/${ anyLangParam }/discover` ], ssrSetupLocale, discoverSsr, makeLayout );
+	if ( isDiscoveryV2Enabled() ) {
+		router(
+			[
+				'/discover',
+				'/discover/add-new',
+				'/discover/firstposts',
+				'/discover/tags',
+				'/discover/reddit',
+				'/discover/latest',
+				`/${ anyLangParam }/discover`,
+				`/${ anyLangParam }/discover/add-new`,
+				`/${ anyLangParam }/discover/firstposts`,
+				`/${ anyLangParam }/discover/tags`,
+				`/${ anyLangParam }/discover/reddit`,
+				`/${ anyLangParam }/discover/latest`,
+			],
+			ssrSetupLocale,
+			discoverSsr,
+			makeLayout
+		);
+	} else {
+		// Original query parameter-based route for v1
+		router(
+			[ '/discover', `/${ anyLangParam }/discover` ],
+			ssrSetupLocale,
+			discoverSsr,
+			makeLayout
+		);
+	}
```

Comment:
It will be better if we refactor the code as following:

```typescript
const routes = [ '/discover', `/${ anyLangParam }/discover` ];

if ( isDiscoveryV2Enabled() ) {
	routes.push(
		'/discover/add-new',
		'/discover/firstposts',
		'/discover/tags',
		'/discover/reddit',
		'/discover/latest',
		`/${ anyLangParam }/discover/add-new`,
		`/${ anyLangParam }/discover/firstposts`,
		`/${ anyLangParam }/discover/tags`,
		`/${ anyLangParam }/discover/reddit`,
		`/${ anyLangParam }/discover/latest`
	);
}

router(
	routes,
	ssrSetupLocale,
	discoverSsr,
	makeLayout
);
```